### PR TITLE
Add rnaquast

### PR DESCRIPTION
**Pull request by lehmann-ju**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Github: [https://github.com/ablab/rnaquast](url)
(can also be added to the quast folder)

### CODE REVIEW

**Comment by bernt-matthias** [comment]
> can also be added to quast

@lehmann-ju means to the quast folder .. potential advantage would be that macros might help in the future

File: tools/rnaquast/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+categories: [Assembly,RNA]
+description: rnaQuast (RNA Quality Assessment Tool) evaluates genome assemblies. 
+remote_repository_url: https://git.ufz.de/lehmanju/rnaquast
+homepage_url: https://github.com/ablab/rnaquast
+name: rnaquast
+owner: lehmanju
```
Comment (bernt-matthias) [inline tools/rnaquast/.shed.yml:6]: The owner should be changed to iuc

File: tools/rnaquast/rna_quast.xml [hunk 2]
```diff
@@ -0,0 +1,453 @@
+<tool id="rna_quast" name="rnaQUAST" version="2.0.1-0" python_template_version="3.5">
+    <description>A Quality Assessment Tool for De Novo Transcriptome Assemblies</description>
+    <requirements>
+        <requirement type="package" version="2.0.1-0">rnaquast</requirement>
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:4]: I think the `-0` is not needed in the version. Also can you use a macro for the version and reuse it in the `<tool>` tag, like here https://github.com/galaxyproject/tools-iuc/blob/4d4576623a85b58137f9e8c5a7747cb2f484c8b6/tools/gprofiler/macros.xml#L11

File: tools/rnaquast/rna_quast.xml [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+    #for $i in $input
+        ln -s '$i' '$i.element_identifier' &&
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:8]: element identifiers need quoting, like here https://github.com/galaxyproject/tools-iuc/blob/4d4576623a85b58137f9e8c5a7747cb2f484c8b6/tool_collections/fastx_toolkit/fastq_quality_boxplot/fastq_quality_boxplot.xml#L9

File: tools/rnaquast/rna_quast.xml [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+    #end for
+    #if $r
+        #for $rf in $r
+            ln -s '$rf' '$rf.element_identifier' &&
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:12]: Also needs quoting

Comment (bernt-matthias) [reply tools/rnaquast/rna_quast.xml:12]: .. and multiple places below

File: tools/rnaquast/rna_quast.xml [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+            <discover_datasets ext="txt" pattern="(?P&lt;name&gt;.+)\.txt"  directory="outputdir/comparison_output/" visible="false" />
+            <filter> len(input)>1 and "comparison" in out_add</filter>
+        </collection>
+        <collection name="data_collection" type="list:list" label="${tool.name} on ${on_string}: detailed output">
+            <discover_datasets pattern="(?P&lt;identifier_0&gt;.+)/(?P&lt;identifier_1&gt;.+)\.(?P&lt;ext&gt;[^.]+)" directory="details/" visible="false"/>
+            <filter> len(input)>1 and "details" in out_add</filter>
+        </collection>
+    </outputs>
+    <tests>
+        <test>
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:131]: can you add the attribute `expect_num_outputs` to check the number of outputs, i.e if the filters work

File: tools/rnaquast/rna_quast.xml [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+    && mkdir details 
+    #for $i in $input
+        #set basename = os.path.splitext(str($i.element_identifier))[0]
+        && mv outputdir/'$basename'_output details/'$basename'
+    #end for
+
+    ]]></command>
+    <inputs>
+        <param name="input" type="data" format="fasta" multiple="true" label="Chromosomes/scaffolds file"/>
+        <param name="strand_specific" argument="-ss" type="boolean" truevalue="-ss" falsevalue="" checked="false" label="Set if transcripts were assembled using strand-specific RNA-Seq data in order to benefit from knowing whether the transcript originated from the + or - strand."/>
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:63]: Can you make the `label` as short as possible, e.g. just "Strand specific", and move everything else to the `help` attribute.

File: tools/rnaquast/rna_quast.xml [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+        <param name="r" optional="true" argument="-r" type="data" format="fasta" multiple="true" label="Reference genome" />
+        <conditional name="gene_coordinates">
+            <param name="use_gtf" type="select" label="Use file with gene coordinates in GTF/GFF format (needs information about parent relations)?" help="We recommend to use files downloaded from GENCODE or Ensembl.">
+                <option value="true" selected="true">Yes</option>
+                <option value="false">No</option>
+            </param>
+            <when value="true">
+                <param name="gtf" argument="--gtf" type="data" format="gtf, gff, gff3" multiple="true" label="GTF/GFF file"/>
+                <param name="disable_infer_genes" argument="--disable_infer_genes" type="boolean" truevalue="--disable_infer_genes" falsevalue="" checked="false" label="Use true if your GTF file already contains genes records, otherwise gffutils will fix it. Note that gffutils may work for quite a long time."/>
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:72]: Also here a shorter label would be nice. Also note that you could remove the name attribute here. Galaxy creates a name equal to the value of `argument` (with leading dashes removed and internal dashed replaced by underscores).

File: tools/rnaquast/rna_quast.xml [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+            </when>
+            <when value="false">
+            </when>
+        </conditional>
+        <param name="prokaryote" argument="--prokaryote" type="boolean" truevalue="--prokaryote" falsevalue="" checked="false" label="Is genome prokararyotic?"/>
+        <param name="min_alignment" argument="--min_alignment" type="integer" value="50" label="Minimal alignment length to be used"/>
+        <param name="blat" argument="--blat" type="boolean" truevalue="--blat" falsevalue="" checked="false" label="Run with BLAT alignment tool instead of GMAP?" />
+        <param name="busco_lineage" argument="--busco_lineage" type="boolean" truevalue="--busco_lineage" falsevalue="" checked="false" label="Run BUSCO tool?" help="The BUSCO tool detects core genes in the assembly. Use this option to provide path to the BUSCO lineage data (Eukaryota, Metazoa, Arthropoda, Vertebrata or Fungi)."/>
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:81]: This label has perfect length :)

File: tools/rnaquast/rna_quast.xml [hunk 9]
```diff
@@ -0,0 +1,453 @@
...
+        <param name="gene_mark" argument="--gene_mark" type="boolean" truevalue="--gene_mark" falsevalue="" checked="false" label="Run with GeneMarkS-T gene prediction tool?"/>
+        <param name="lower_threshold" argument="--lower_threshold" type="integer" value="50" label="Lower threshold for x_assembled/covered/matched metrics."/>
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:83]: Are there minima or maxima for the integer parameters?

File: tools/rnaquast/rna_quast.xml [hunk 10]
```diff
@@ -0,0 +1,453 @@
...
+        <param name="upper_threshold" argument="--upper_threshold" type="integer" value="95" label="Upper threshold for x_assembled/covered/matched metrics."/>
+        <param name="out_sr" type="select" multiple="true" label="Short report formats">
+            <option value="tsv" selected="true">tabular</option>
+            <option value="txt">txt</option>
+            <option value="tex">tex</option>
+            <option value="pdf" selected="true">pdf</option>
+        </param>
+        <param name="out_add" type="select" multiple="true" label="Additional outputs">
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:91]: Is this optional?

File: tools/rnaquast/rna_quast.xml (hunk 9 above)
Comment (lehmann-ju) [reply tools/rnaquast/rna_quast.xml]: The value is given in in percent. Probably it have to be between 0 and 100.

Comment (bernt-matthias) [reply tools/rnaquast/rna_quast.xml]: You checked in the sources?

File: tools/rnaquast/rna_quast.xml [hunk 11]
```diff
@@ -4,37 +4,44 @@
         <requirement type="package" version="2.0.1">rnaquast</requirement>
     </requirements>
     <command detect_errors="exit_code"><![CDATA[
+    #import re
     #for $i in $input
-        ln -s '$i' '$i.element_identifier' &&
+        #set input_element_identifier = re.sub('[^\w\-\s]', '_', str($i.element_identifier))
+        ln -s '$i' '$i.input_element_identifier' &&
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:10]: Now you only need to use `input_element_identifier` :) Or simply `ln -s '$i' '${re.sub('[^\w\-_.]', '_', f.element_identifier)}' &&`

File: tools/rnaquast/rna_quast.xml (hunk 9 above)
Comment (lehmann-ju) [reply tools/rnaquast/rna_quast.xml]: There is no minima or maxima in the sources.

Comment (bernt-matthias) [reply tools/rnaquast/rna_quast.xml]: OK

**Comment by bernt-matthias** [comment]
Seems that there is test data missing `idba.fasta` and `spades.311.fasta`

**Comment by bernt-matthias** [comment]
Problem seems to be that the test data folder needs to be named `test-data`

**Comment by bernt-matthias** [comment]
Seems that the collections remain empty. 

Also in the tests the collection elements need to be sorted alphabetically.

File: tools/rnaquast/rna_quast.xml [hunk 12]
```diff
@@ -0,0 +1,462 @@
...
+            <output_collection name="details_png" type="list:list" count="1">
+                <element name="spades.311">
+                    <element_has_text name="Nx" text="PNG"/>
+                    <element_has_text name="transcript_length" text="PNG"/>
+                </element>
+            </output_collection>
+        </test>
+    </tests>
+    <help><![CDATA[
+
```
Comment (bgruening) [inline tools/rnaquast/rna_quast.xml:249]: That tools looks already cool! Thanks a lot!

To improve the help a little bit you can use http://rst.ninjs.org to see how it looks like. There is some information included that is not needed in the Galaxy context, maybe you can make the help more concise?

**Comment by bernt-matthias** [comment]
Currently tests do not run because of: 

```
Exception: Test output defines nothing to check (e.g. must have a 'file' check against, assertions to check, metadata or checksum tests, etc...)
```

Seems to be a bug in the CI framework that the tests report as sucessful.

File: tools/rnaquast/rna_quast.xml [hunk 13]
```diff
@@ -204,10 +204,7 @@
             <output name="short_report_txt" file="short_report.txt" compare="sim_size"/>
             <output name="short_report_tex" file="short_report.tex" compare="sim_size"/>
             <output name="short_report_tsv" file="short_report.tsv" compare="sim_size"/>
-            <output_collection name="list_logs" type="list">
-                <element name="rnaQUAST" file="rnaQUAST"/>
-                <element name="Trinity.GeneMarkS_T.err" file="spades.311.GeneMarkS_T.err"/>
-            </output_collection>
+            <output_collection name="list_logs" type="list" coutn="2"/>
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:207]: coutn -> count

File: tools/rnaquast/rna_quast.xml [hunk 14]
```diff
@@ -0,0 +1,289 @@
...
+            <output name="short_report_tsv">
+                <assert_contents>
+                    <has_line_matching expression="^METRICS/TRANSCRIPTS\tidba\tTrinity$"/>
+                </assert_contents>
+            </output>
+            <output_collection name="comparison_png" type="list" count="4"/>
+            <output_collection name="comparison" type="list" count="12"/>
+            <output_collection name="list_logs" type="list" count="3"/>
+            <output_collection name="details" type="list:list" count="2">
+                <output_collection name="Trinity" type="list" count="11"/>
```
Comment (bernt-matthias) [inline tools/rnaquast/rna_quast.xml:189]: Can you make indentation consistent (4 spaces for each indentation level)?

Comment (bernt-matthias) [reply tools/rnaquast/rna_quast.xml:189]: nesting output_collection tags is not allowed. there it should be 1 or more nested element tags within the output_collection (as before). 

Also I would ask you to restore the assertions...

Comment (lehmann-ju) [reply tools/rnaquast/rna_quast.xml]: With this nesting output_collections the test could run again.
I removed them but now the old Exception which you had mentioned yesterday was thrown again.

File: tools/rnaquast/rna_quast.xml (hunk 12 above)
Comment (lehmann-ju) [reply tools/rnaquast/rna_quast.xml]: Thank you! 
Hopefully it is better now?

**Comment by bernt-matthias** [comment]
Nearly threre :) .. I will finish my part this afternoon

**Comment by bernt-matthias** [comment]
@lehmann-ju I'm done here. Also reduced the test data a bit more. 

Attention I force pushed to the branch (I squashed the commits). If you want to continue:

```
git checkout rnaquast
git fetch --all
git reset --hard origin/rnaquast
```

(assuming that your remote is called origin)

**Comment by bernt-matthias** [comment]
Hmm. odd planemo test works locally for me ..

**Comment by bernt-matthias** [comment]
@bgruening I guess the container is broken: 

`makeblastdb: error while loading shared libraries: libbz2.so.1: cannot open shared object file: No such file or directory`

tests using conda envs work.

**Comment by bernt-matthias** [comment]
hurray .. linter problem disappeared

**Comment by bernt-matthias** [comment]
magic :)

**Comment by bgruening** [comment]
Can we try to fix the container instead? I have started it here: https://github.com/bioconda/bioconda-recipes/pull/24890

**Comment by bernt-matthias** [comment]
> Can we try to fix the container instead?

indeed

@lehmann-ju can you undo the changed to `.tt_biocontainers_skip`?

**Review by bernt-matthias — APPROVED** [review]

**Pull request merged**